packages/firestore/src/local/simple_db.ts

```diff
@@ -263,19 +260,30 @@ export class SimpleDbTransaction {
   }
 
   constructor(private readonly transaction: IDBTransaction) {
-    this.completionPromise = new Promise<void>((resolve, reject) => {
-      // We consider aborting to be "normal" and just resolve the promise.
-      // May need to revisit if/when we actually need to abort transactions.
-      this.transaction.onabort = this.transaction.oncomplete = event => {
-        resolve();
-      };
-      this.transaction.onerror = (event: Event) => {
-        reject((event.target as IDBRequest).error);
-      };
-    });
+    this.completionPromise = new Deferred<void>();
+
+    this.transaction.oncomplete = () => {
+      this.completionPromise.resolve();
+    };
+    this.transaction.onabort = () => {
+      // TODO(mrschmidt): If in the future we end up adding code that
+      // intentionally aborts transactions we may need to turn this into a
+      // non-error case.
+      const error =
+        transaction.error ||
+        new Error('The IndexedDb transaction was aborted.');
+      this.completionPromise.reject(error);
```

Comment:
So actually...  could we change this to:

```
if (transaction.error) {
  this.completionPromise.reject(error);
}
```
?

I think that would mean:
1. You can call transaction.abort() [no error] to gracefully abort the transaction but not trigger any error handling.
2. It would solve a wart where as-is, I think if you call transaction.abort(error), then we'll end up calling completionPromise.reject() in abort() and then again here which seems sloppy / confusing.

Reply:
Done